airflow/providers/google/cloud/operators/dataplex.py

```diff
@@ -0,0 +1,395 @@
+    :type body: Dict[str, Any]
+    :param dataplex_task_id: Required. Task identifier.
+    :type dataplex_task_id: str
+    :param validate_only: Optional. Only validate the request, but do not perform mutations. The default is
+        false.
+    :type validate_only: bool
+    :param api_version: The version of the api that will be requested for example 'v3'.
+    :type api_version: str
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    :param asynchronous: Flag informing should the Dataplex task be created asynchronously.
+        This is useful for long running creating tasks and
+        waiting on them asynchronously using the DataplexTaskSensor
+    :type asynchronous: bool
+    """
+
+    template_fields = (
+        "project_id",
+        "dataplex_task_id",
+        "body",
+        "validate_only",
+        "delegate_to",
+        "impersonation_chain",
+    )
+    template_fields_renderers = {'body': 'json'}
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        lake_id: str,
+        body: Dict[str, Any],
+        dataplex_task_id: str,
+        validate_only: Optional[bool] = None,
+        api_version: str = "v1",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: str = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        asynchronous: bool = False,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.lake_id = lake_id
+        self.body = body
+        self.dataplex_task_id = dataplex_task_id
+        self.validate_only = validate_only
+        self.api_version = api_version
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+        self.asynchronous = asynchronous
+
+    def execute(self, context: dict) -> dict:
```

Comment:
Now that 2.2.3 has been released, typing for `context` can be:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from airflow.utils.context import Context
...

def execute(self, context: "Context") -> dict:
...
```
This can be applied to all of the operators too.

Reply:
TIL 🚀 Thanks @josh-fell

Reply:
Yeah. I applied it globally during the Xmas break. Just wonder. Maybe we should add a pre-commit checking if the "old ways" are still used. WDYT @turbaszek @josh-fell ?

Reply:
+1 Definitely worth automating.